code/modules/mob/living/living.dm

```diff
@@ -600,6 +601,61 @@
 		return 1
 	return 0
 
+/mob/living/proc/makeTrail(turf/T)
+	if(!has_gravity(src))
+		return
+	var/blood_exists = 0
+	var/mob/living/carbon/human/H = src
+
+	if(H.species.exotic_blood)
+		return
+
+	for(var/obj/effect/decal/cleanable/trail_holder/C in loc) //checks for blood splatter already on the floor
+		blood_exists = 1
+	if(isturf(src.loc))
+		var/trail_type = getTrail()
+		if(trail_type)
+			var/brute_ratio = round(getBruteLoss()/maxHealth, 0.1)
+			if(blood_volume && blood_volume > max(BLOOD_VOLUME_NORMAL*(1 - brute_ratio * 0.25), 0))//don't leave trail if blood volume below a threshold
+				blood_volume = max(blood_volume - max(1, brute_ratio * 2), 0) 					//that depends on our brute damage.
+				var/newdir = get_dir(T, src.loc)
+				if(newdir != src.dir)
+					newdir = newdir | src.dir
+					if(newdir == 3) //N + S
+						newdir = NORTH
+					else if(newdir == 12) //E + W
+						newdir = EAST
+				if((newdir in cardinal) && (prob(50)))
+					newdir = turn(get_dir(T, src.loc), 180)
+				if(!blood_exists)
+					new /obj/effect/decal/cleanable/trail_holder(loc)
+				for(var/obj/effect/decal/cleanable/trail_holder/TH in loc)
+					if((!(newdir in TH.existing_dirs) || trail_type == "trails_1" || trail_type == "trails_2") && TH.existing_dirs.len <= 16) //maximum amount of overlays is 16 (all light & heavy directions filled)
+						TH.existing_dirs += newdir
+						TH.overlays.Add(image('icons/effects/blood.dmi', trail_type, dir = newdir))
+						TH.transfer_mob_blood_dna(src)
+						if(istype(H) && H.species.blood_color)
+							TH.color = H.species.blood_color
+
+/mob/living/carbon/human/makeTrail(turf/T)
+	if(species.exotic_blood && bleed_rate)
+		if(!T)
+			T = get_turf(src)
+		var/datum/reagent/R = chemical_reagents_list[get_blood_id()]
+		if(istype(R))
+			R.reaction_turf(T, 10)
```

Comment:
This is going to be unholy levels of bullshit with slime blood.

Every tile they're dragged across, it's going to put make the tile slippery.